reconcile/quay_permissions.py

```diff
@@ -77,48 +78,54 @@ def run(dry_run):
                 logging.debug(['repo', repo_name])
 
                 for team in teams:
-                    permissions = team['permissions']
-                    role = team['role']
-                    for permission in permissions:
-                        if permission['service'] != 'quay-membership':
-                            logging.warning('wrong service kind, '
-                                            'should be quay-membership')
-                            continue
-
-                        perm_org_key = (
-                          permission['quayOrg']['instance']['name'],
-                          permission['quayOrg']['name']
-                        )
-
-                        if perm_org_key != org_key:
-                            logging.warning('wrong org, should be %s', org_key)
-                            continue
-
-                        team_name = permission['team']
-
-                        # processing team section
-                        logging.debug(['team', team_name])
-
-                        current_role = \
-                            quay_api.get_repo_team_permissions(
-                                repo_name, team_name)
-                        if current_role != role:
-                            logging.info(
-                                ['update_role', org_key, repo_name,
-                                 team_name, role])
-                            if not dry_run:
-                                try:
+                    try:
```

Comment:
This `try` block is either navigating through dicts or doing calls. I don't get understand need for such a broad scope. Could we just start it in `get_repo_team_permissions`?

Reply:
The goal here is to do as much work as possible even if some repositories or teams fail.

If we can't even get repo permissions we can't continue the work of the iteration. Just having `try: except: log;continue` is more and harder-to-read code. Just having the happy path inside the `try` block leads to less code, and makes sure any other mistake I may have forgotten to consider is handled. This is the top-level in which we want unexpected exceptions handled and reported grafecully.